clientcmd: ConfirmUsable panics on invalid kubeconfig

Repro steps: Create a .kube/config with a "user" pointing to a user that doesn't exist.

Expected result: Get a validation error

Actual result:
There are two different ConfirmUsable methods, and they have different behaviior.

The first one:
https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/client_config.go#L395
discards the error, and continues successfully. Downstream, kubectl will prompt for a username.

The second one:
https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/validation.go#L182
panics with this stack trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5010180]

goroutine 1 [running]:
k8s.io/client-go/tools/clientcmd.ConfirmUsable({{0x0, 0x0}, {0x0, 0x0}, {0x0, 0xc00055a750}, 0xc00055a780, 0xc00055a7b0, 0xc00055a7e0, {0xc000172da0, ...}, ...}, ...)
        k8s.io/client-go@v0.23.6/tools/clientcmd/validation.go:207 +0x480
github.com/tilt-dev/tilt/internal/k8s.ProvideAPIConfig({0x6f5d2a0, 0xc000520a00}, {0x0, 0x0}, {0x0, 0x0})
        github.com/tilt-dev/tilt/internal/k8s/env.go:46 +0x218
```

What's the right behavior here? I'm happy to send a fix but unclear what the right behavior is.

Original bug report: https://github.com/tilt-dev/tilt/issues/5831 (we don't know how the user got an invalid config, we're several steps removed from the original bug report)

--- a/staging/src/k8s.io/client-go/tools/clientcmd/validation.go
+++ b/staging/src/k8s.io/client-go/tools/clientcmd/validation.go
@@ -204,8 +204,19 @@ func ConfirmUsable(config clientcmdapi.Config, passedContextName string) error {
 
 	if exists {
 		validationErrors = append(validationErrors, validateContext(contextName, *context, config)...)
-		validationErrors = append(validationErrors, validateAuthInfo(context.AuthInfo, *config.AuthInfos[context.AuthInfo])...)
-		validationErrors = append(validationErrors, validateClusterInfo(context.Cluster, *config.Clusters[context.Cluster])...)
+
+		// Default to empty users and clusters and let the validation function report an error.
+		authInfo := config.AuthInfos[context.AuthInfo]
+		if authInfo == nil {
+			authInfo = &clientcmdapi.AuthInfo{}
+		}
+		validationErrors = append(validationErrors, validateAuthInfo(context.AuthInfo, *authInfo)...)
+
+		cluster := config.Clusters[context.Cluster]
+		if cluster == nil {
+			cluster = &clientcmdapi.Cluster{}
+		}
+		validationErrors = append(validationErrors, validateClusterInfo(context.Cluster, *cluster)...)
 	}
 
 	return newErrConfigurationInvalid(validationErrors)

--- a/staging/src/k8s.io/client-go/tools/clientcmd/validation_test.go
+++ b/staging/src/k8s.io/client-go/tools/clientcmd/validation_test.go
@@ -65,6 +65,42 @@ func TestConfirmUsableBadInfoButOkConfig(t *testing.T) {
 	badValidation.testConfig(t)
 }
 
+func TestConfirmUsableMissingObjects(t *testing.T) {
+	config := clientcmdapi.NewConfig()
+	config.Clusters["kind-cluster"] = &clientcmdapi.Cluster{
+		Server: "anything",
+	}
+	config.AuthInfos["kind-user"] = &clientcmdapi.AuthInfo{
+		Token: "any-value",
+	}
+	config.Contexts["missing-user"] = &clientcmdapi.Context{
+		Cluster:  "kind-cluster",
+		AuthInfo: "garbage",
+	}
+	config.Contexts["missing-cluster"] = &clientcmdapi.Context{
+		Cluster:  "garbage",
+		AuthInfo: "kind-user",
+	}
+
+	missingUser := configValidationTest{
+		config: config,
+		expectedErrorSubstring: []string{
+			`user "garbage" was not found for context "missing-user"`,
+		},
+	}
+	missingUser.testConfirmUsable("missing-user", t)
+	missingUser.testConfig(t)
+
+	missingCluster := configValidationTest{
+		config: config,
+		expectedErrorSubstring: []string{
+			`cluster "garbage" was not found for context "missing-cluster"`,
+		},
+	}
+	missingCluster.testConfirmUsable("missing-cluster", t)
+	missingCluster.testConfig(t)
+}
+
 func TestConfirmUsableBadInfoConfig(t *testing.T) {
 	config := clientcmdapi.NewConfig()
 	config.Clusters["missing ca"] = &clientcmdapi.Cluster{

